packages/react/src/status-badge/StatusBadge.tsx

```diff
@@ -55,54 +120,167 @@ export const StatusBadge = ({
+		iconLabel: 'information, medium emphasis',
+	},
+	inProgressLow: {
+		borderColor: 'border',
+		icon: ProgressDoingIcon,
+		iconColor: 'muted',
+		iconLabel: 'in, low emphasis progress',
+	},
+	notStartedLow: {
+		borderColor: 'border',
+		icon: ProgressTodoIcon,
+		iconColor: 'muted',
+		iconLabel: 'not started, low emphasis',
+	},
+	pausedLow: {
+		borderColor: 'border',
+		icon: ProgressPausedIcon,
+		iconColor: 'muted',
+		iconLabel: 'paused, low emphasis',
+	},
+	successHigh: {
+		borderColor: 'success',
+		icon: SuccessFilledIcon,
+		iconColor: 'success',
+		iconLabel: 'success, high emphasis',
+	},
+	successLow: {
+		borderColor: 'border',
+		icon: SuccessIcon,
+		iconColor: 'muted',
+		iconLabel: 'success, low emphasis',
+	},
+	successMedium: {
+		borderColor: 'success',
+		icon: SuccessIcon,
+		iconColor: 'success',
+		iconLabel: 'success, medium emphasis',
+	},
+	unknownLow: {
+		borderColor: 'border',
+		icon: HelpIcon,
+		iconColor: 'muted',
+		iconLabel: 'help, low emphasis',
+	},
+	warningHigh: {
+		borderColor: 'warning',
+		icon: WarningFilledIcon,
+		iconColor: 'warning',
+		iconLabel: 'warning, high emphasis',
+	},
+	warningLow: {
+		borderColor: 'border',
+		icon: WarningCircleIcon,
+		iconColor: 'muted',
+		iconLabel: 'warning, low emphasis',
+	},
+	warningMedium: {
+		borderColor: 'warning',
+		icon: WarningIcon,
+		iconColor: 'warning',
+		iconLabel: 'warning, medium emphasis',
+	},
+} as const;
+
 export type StatusBadgeTone = keyof typeof toneMap;
+export type StatusBadgeAppearance = 'regular' | 'subtle';
 
-const NeutralDot = () => (
-	<Box
-		highContrastOutline
-		css={{
-			width: 8,
-			height: 8,
-			borderRadius: 4,
-			backgroundColor: boxPalette.foregroundMuted,
-		}}
-	/>
-);
+// Deprecated Legacy
+export type StatusBadgeLegacyTone = keyof typeof legacyToneMap;
 
-const toneMap = {
+const legacyToneMap = {
 	neutral: {
-		borderColor: boxPalette.border,
-		icon: () => <NeutralDot />,
+		icon: () => (
+			<Box
+				highContrastOutline
+				css={{
+					width: 8,
+					height: 8,
+					borderRadius: 4,
+					backgroundColor: boxPalette.foregroundMuted,
+				}}
+			/>
+		),
+		tone: 'border',
 	},
 	success: {
-		borderColor: boxPalette.systemSuccess,
 		icon: () => (
 			<SuccessIcon color="success" aria-hidden="false" aria-label="Success" />
 		),
+		tone: 'success',
 	},
 	error: {
-		borderColor: boxPalette.systemError,
 		icon: () => (
 			<AlertIcon color="error" aria-hidden="false" aria-label="Error" />
 		),
+		tone: 'error',
 	},
-
 	info: {
-		borderColor: boxPalette.systemInfo,
 		icon: () => (
 			<InfoIcon color="info" aria-hidden="false" aria-label="Information" />
 		),
+		tone: 'info',
 	},
 	warning: {
-		borderColor: boxPalette.systemWarning,
 		icon: () => (
 			<WarningIcon color="warning" aria-hidden="false" aria-label="Warning" />
 		),
+		tone: 'warning',
 	},
 } as const;
 
-export type StatusBadgeWeight = 'subtle' | 'regular';
+const legacyToneMapKeys = Object.keys(legacyToneMap);
```

Comment:
`Object.keys(obj)` returns as type `Array<string>`

This should be typed as `Array<keyof typeof legacyToneMap>` so that it is easy to use the array values to look up values in `legacyToneMap`.

The `getKeyValues(obj)` utility function in Component Catalogue was made for this purpose.

Reply:
Adding `as Array<StatusBadgeLegacyTone>` causes more issues than it solves, because then `const isLegacy = legacyToneMapKeys.includes(tone)` on L41 errors because it has tones it's not allowed to have.

I could work around this by making it `as Array<StatusBadgeLegacyTone | StatusBadgeTone>;` but that's clearly wrong, or I could write a function to handle it, but I do not want to write additional code that users have to download just to appease TS, especially when this is all for legacy tones that will never be updated.

Reply:
Fair points. We can leave it as `Object.keys`.